File: esprit-internship2-front/src/models/Container.ts
import { Containment } from './Containment';
import { Station } from './Station';

export interface Container {
  id?: number;
  title?: string;
  creationDate?: string;
  photo?: string;
  weight?: number;
  weightLimit?: number;
  width?: number;
  length?: number;
  height?: number;
  cost?: number;

  containerType?: ContainerType;
  station?: Station; 

  containment?: Containment; 

  state?: number;
}

export enum ContainerType {
  Metallic = 'Metallic',
  Plastic = 'Plastic',
  Wooden = 'Wooden',
  Cardboard = 'Cardboard',
  Special = 'Special'
}

File: esprit-internship2-back/src/main/java/tn/esprit/espritinternship2back/entity/Container.java
package tn.esprit.espritinternship2back.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Container {

    @Id
    @GeneratedValue
    private Long id;
    private String title;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String photo = "";
    private Float weight;
    private Float weightLimit;
    private Float width;
    private Float length;
    private Float height;
    private Float cost;

    @Enumerated(EnumType.STRING)
    private ContainerType containerType;
    @Enumerated(EnumType.STRING)
    private Station station;

    private int state = 0;

    public enum ContainerType {
        Metallic,
        Plastic,
        Wooden,
        Cardboard,
        Special
    }

}
